src/index.js

```diff
@@ -260,11 +267,16 @@ class WarmUP {
     /** Log warmup start */
     this.serverless.cli.log('WarmUP: setting ' + functionNames.length + ' lambdas to be warm')
 
-    /** Get function names */
-    functionNames = functionNames.map((functionName) => {
+    /** Get necessary function info */
+    let functions = functionNames.map((functionName) => {
+      let functionInfo = {}
```

Comment:
functionObject.warmup is either a bool, a single stage or an array of stages according to the documentation and how the enable function checks it in createWarmer.

If you try and use any of that along with `functionObject.warmup.conccurrency` (or any other child setting) you get a YAML indentation exception.  So if you wanted to have a function with selected stages and a custom concurrency level then using `functionObject.warmup.concurrency` won't work.

Perhaps we could move the `functionObject.warmup` setting to `functionObject.warmup.enable` and update the documentation so that going forward that's used.  For people already using this plugin  it's easy enough to check if `functionObject.warmup` if `functionObject.warmup.enable` or `functionObject.warmup.concurrency` aren't found.

Making that change would also allow for future function-specific options for warmup.